Sources/AWSSDKSwiftCore/Message/AWSRequest.swift

```diff
@@ -326,8 +283,26 @@ extension AWSRequest {
         self.httpMethod = httpMethod
         self.httpHeaders = headers
         self.body = body
+        
+        addStandardHeaders()
     }
-    
+
+    private mutating func addStandardHeaders() {
+        if httpHeaders["content-type"].first == nil {
+            switch httpMethod {
+            case "GET","HEAD":
+                break
+            default:
+                if case .restjson = serviceProtocol, case .raw(_) = body {
+                    httpHeaders.replaceOrAdd(name: "content-type", value: "binary/octet-stream")
+                } else {
+                    httpHeaders.replaceOrAdd(name: "content-type", value: serviceProtocol.contentType)
+                }
+            }
+        }
+        httpHeaders.replaceOrAdd(name: "user-agent", value: "AWSSDKSwift/5.0")
```

Comment:
```suggestion
        httpHeaders.replaceOrAdd(name: "user-agent", value: "AWSSDKSwift/5.0")
        guard httpHeaders["content-type"].first == nil else {
            return
        }
        
        guard httpMethod != "GET", httpMethod != "POST" else {
            return
        }

        if case .restjson = serviceProtocol, case .raw(_) = body {
            httpHeaders.replaceOrAdd(name: "content-type", value: "binary/octet-stream")
        } else {
            httpHeaders.replaceOrAdd(name: "content-type", value: serviceProtocol.contentType)
        }
```

Let's see if we can make this code look a little less complex

Reply:
Nice typo, in your alternative 
```
guard httpMethod != "GET", httpMethod != "POST" else
```
I think you meant "HEAD"